web/web/user_settings/static/actions.js

```diff
@@ -0,0 +1,59 @@
+import { createErrorMessage } from "../../static/js/helpers";
+import { api } from "../../static/js/network_client";
+
+export function updateSuccessful() {
+    return {
+        type: "UPDATE_SUCCESSFUL",
+    };
+}
+
+export function updateFailed() {
+    return {
+        type: "UPDATE_FAILED",
+    };
+}
+
+export function updateUserSettings(
+    userId,
+    username,
+    password,
+    firstName,
+    lastName
+) {
+    return (dispatch) => {
+        try {
+            const userData = {
+                user: {
+                    id: userId,
+                    first_name: firstName,
+                    last_name: lastName,
+                },
+                login: {
+                    username: username,
+                    password_hash: password,
+                },
+            };
+            api.patch(
+                "update_user_settings",
+                { json: { ...userData } },
+                (data) => {
+                    dispatch(updateSuccessful());
+                    createErrorMessage(
```

Comment:
Why are we using an error message function to display a success notification? I think we should either rename the function to be more semantically correct or create a success modal.

Reply:
done